### ISSUE TITLE
Visuals not matching osu!stable

### ISSUE THREAD

**Issue by Reco1I** [osudroid/osu-droid#333]
We know osu!droid is based on osu!stable but it doesn't look like it, there's things missing or looking different that we should take in consideration:
 
### Globally:
* Android toast should be replaced with a game rendered toast:
![image](https://github.com/osudroid/osu-droid/assets/103213881/5ee483c1-b7f8-43ff-a4ba-6eb9b361fec2)
* Cursors should be visible on every screen.
 
### Intro screen
* Loading icon and loading process should be removed, consider showing the osu! logo instead.
* Intro animation and main menu animation should be skipped if there's pending beatmaps to import.
* When showing the "_Welcome_" sprite, the logo spectrum should be fade in blue tinted:
![image](https://github.com/osudroid/osu-droid/assets/103213881/60c9ca27-dab9-448c-b0f7-ed0d42d3d253)

### Main screen
 * Logo menu should have eased animations when showing/hiding the menu.
 * Logo menu should have eased animations when navigating to second set of buttons (for example when pressing play).
 * Logo menu buttons should be purple and once clicked tinted to pink with a fade animation.
 * Logo spectrum should have a shifting animation based on BPM:
 
https://github.com/osudroid/osu-droid/assets/103213881/7663845b-9642-436f-8968-95221b43e936
 
 * Logo should have ripples following the beat:

https://github.com/osudroid/osu-droid/assets/103213881/4c44ddf5-9454-4038-b46d-f6b5bd8b2549

 * The screen should have a bottom bar with the same size of the top bar.
 * The screen should have light shadows on both horizontal sides following the beat.
 * Player card should have an "osu!std" icon on the top right side.
 * Transition to song menu should have a fade out animation.
 
 ### Main screen Now playing layout
  * The title should auto collapse fading out.
  * Additionally an "_Show info_" button should be added:
![image](https://github.com/osudroid/osu-droid/assets/103213881/18656e72-6e10-4712-a03f-0ef87f8c7458)
 * A "_Jump to_" button should be added, when pressed it should show the following screen:
![image](https://github.com/osudroid/osu-droid/assets/103213881/d67827ac-13cc-4936-a230-afd1ffe73406)

### Settings menu
* It should be an left side panel:
![image](https://github.com/osudroid/osu-droid/assets/103213881/c2ef42a5-6db4-4138-8499-b742dd9d049b)
* It should show a preview of the current selected skin:
![image](https://github.com/osudroid/osu-droid/assets/103213881/2757ee0c-c5c4-4365-8297-def9b2baa2bb)
* Every list selector should be replaced with a dropdown.
* When changing skin there should be a fade out-in animation.

### Loading screen to song menu
 * It should be shown only if there's beatmaps to be imported.
 * "_Your beatmaps are being processed_" sprite should not be skinnable.
 * The background must be the same as the current song background.
 
### Song menu
* When transitioning into the screen there sould be a fade in animation, song list and leaderboard should have a translation animation from its anchors (left for leaderboard and right for song list).

### Song menu top bar
 * The background sprite should have 100% opacity.
 * The background should be skinnable.
 * The left side should have an icon indicating the ranking status of the beatmap or rather nothing (in case the status is unknown).
 * The beatmap details text should have the following layout:
![image](https://github.com/osudroid/osu-droid/assets/103213881/fc29dc5d-e335-4c53-a7d5-45ad3951460a)
 * Below the beatmap details it should be a dropdown to switch from global to local leaderboards (and possible other leaderboards).
 * Right to the leaderboard dropdown there should be a button linking to the selected beatmap site (Chimu or bancho):
![image](https://github.com/osudroid/osu-droid/assets/103213881/9280596e-b73c-4e99-b7cc-b74fd23a925c)
 * "_Sort_" option should be a dropdown option in the right side of the top bar:
![image](https://github.com/osudroid/osu-droid/assets/103213881/eef97ae4-7ded-4246-8717-14f0401a1029)
* Additionally left to the "_Sort_" dropdown there should be a second dropdown for the "_Group_" option:
![image](https://github.com/osudroid/osu-droid/assets/103213881/6a25c16a-d300-4d7e-8e05-f7d58ec2cb74)
* Below "_Sort_" and "_Group_" dropdowns there should be a 4 buttons horizontal layout corresponding to 4 combinations of group and sort options values:
![image](https://github.com/osudroid/osu-droid/assets/103213881/0c328d9a-8c6a-406d-aeff-6867c2dcc470)
* Search bar should be on the top bar.

### Song menu leaderboard
* When loading it should show a loading icon rather than a text.
* Once loaded it should have a sequential bounce animation on every score card.
* Personal best should have fixed position at the bottom.
* When no scores are set this sprite should be shown:
![image](https://github.com/osudroid/osu-droid/assets/103213881/e633d30f-7909-460c-8123-8ff374c80277)
* Scores cards should have a sequential lighten animation: 

https://github.com/osudroid/osu-droid/assets/103213881/f341228e-4b41-4a86-ab0b-edaefb6fc288

### Song menu songs list
* There should be an "osu!std" logo (in osu!stable it depends of the selected mode) in the background with ligthen animation following the beat, when the beatmap is started it expands and fades out before showing loading screen:

https://github.com/osudroid/osu-droid/assets/103213881/c51e9ba4-28db-403e-bc86-fed0ebdbe899
* Song cards carrousel effect should be eased.
* Song card selection should be animated.
* Song cards should have 3 states of tint defined by skin, as an example the default skin has:

| Unactive set | Active set | Active song |
|--------------|------------|-------------|
| Pink         | Blue       | White       |

![image](https://github.com/osudroid/osu-droid/assets/103213881/50b6387a-0167-4a0b-94cb-9b08a462e317)

* Song card text should have the following layout:
![image](https://github.com/osudroid/osu-droid/assets/103213881/d6fb4ad4-a1e0-4a48-9702-3b5051c482d7)
* Additionally there should be an option to show a background preview on every card.
* Selected song card should have an stars particles animation behind:

https://github.com/osudroid/osu-droid/assets/103213881/a9f4e44f-bfd1-4b47-968b-63cacf5693bd
* Scrollbar should be lighter and a dark background.

### Song bottom bar
* The bottom bar should have a 100% opacity background.
* The buttons should have a fade animations between states (pressed down and up)
* The buttons layout should be:

| Mods | Random | Options |
| -- | -- | -- |
* The beatmap options should be accesible from that third button along with long pressing the song card, this menu should have an fade in-out and boucing buttons animations when in transitions.
* On the right side there should be a logo with a bouncing animation following the beat, when pressed this should start the beatmap:

https://github.com/osudroid/osu-droid/assets/103213881/4c716857-e601-44b3-8aed-15292c461e7f

### Mod menu
* There should be a clarifying text at the start of every mods column:
![image](https://github.com/osudroid/osu-droid/assets/103213881/87c6296d-1fa2-4c7b-92d0-9f929ec93442)
* "_Reset all mods_" and "_Close_" buttons should be one aligned vertically in the center of the screen.
* Buttons selection and score multiplier text color change should be animated.
* Transitions to the menu should be animated with fade in-out animations a boucing animation in bottom buttons.

### Loading screen to gameplay
* There should be a "_Visual settings_" layout:
![image](https://github.com/osudroid/osu-droid/assets/103213881/ca74f56c-4ea1-4d44-b51b-b3b2ee9dda9e)
* The background must be the same as the song background.
* If the beatmap is loved or its marked as having "_rapidly flashing colors_" it should shown a disclamer before starting gameplay.

### Gameplay
* The beatmap progress indicator should be a radial progress bar:
![image](https://github.com/osudroid/osu-droid/assets/103213881/bf03ff26-a201-4fcb-bfb4-bb7ba87aaa65)
* The health bar should clip from the right rather than stretching its width.
* Slider tick points sprites should be shown in sequence with an scale-in animation once the slider finish its snaking-in animation.
* Follow points should be placed every 32 pixels until end of segment as described in #115 
* Sliders should have a body with gradient coloring.
* Slider body alpha should be always 0.7
* Hit circle combo number should be always scaled 0.8
* Hit circles should have a shaking animation when trying to hit them before time.
* Hit circles should have a fade out animation when missed.
* Miss sprite should have a falling animation.
* Hit circles should appear with slightly dark color and light up once they're ready to be clicked.
* When failing all objects should have a failing animation.
* When retrying a text "_`x` retries and counting_" text should be shown.

### Scoring scene
* The screen should be scrollable showing 3 sections below the play statistics, "_Local_", "_Online_" and "_Beatmap_":
![image](https://github.com/osudroid/osu-droid/assets/103213881/dd674661-bea5-4e2b-97d7-4dcca543b459)
* When transitioning to the screen there should be a fade in animation and every text of the play statistics should have a fade out and  translation from left side animation.

More examples and items will be added.

**Comment by CELESTYNAI** [comment]
I think you're asking too much for a small team...

plus, for ["Your beatmaps are being processed" sprite should not be skinnable.], I don't think it should be NOT skinnable.

**Comment by Rian8337** [comment]
If you see the PR above, you can see that he worked on it by himself.

**Comment by Reco1I** [comment]
Closing this because after few consideration I concluded that this shouldn't be implemented.

First of all we can't use ppy assets, so this will require to rework some assets to make the similar not equal to the original ones.
Also there's some conflictive UI parts that cannot be ported/paired with osu!stable because they will not be comfortable for mobile devices.

This should be rethink into a different approach.

Consider #352 and #346 as a beginning of it.
